### PR TITLE
Add a Markdown how-to

### LINKED ISSUE #128: Markdown output support?
Hello,

I was wondering if a bit more support for Markdown might be possible? I've more or less successfully ~tricked~ *convinced* towncrier to output Markdown for my changelog file, however there's no way to change or enforce the H1 title output in this case. Right now to sort of get around it, I've done the following in my pyproject.toml:

```toml
[tool.towncrier]
package = 'dripstone'
package_dir = 'src'
filename = 'CHANGELOG.md'
directory = ".changelog"
template = ".changelog/template.md"
title_format = "# {name} {version} ({project_date})"
issue_format = "#{issue}"
start_string = "<!-- TOWNCRIER -->\n"
underlines = ["<!--- --->", "##", "###"]
```

This (plus a slight change to the default template to make it more markdown friendly) then gets generated into

```markdown
# Dripstone 0.1.0 (2018-09-06)
<!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- ---><!--- --->

<!-- TOWNCRIER -->


## Bugfixes

* Fixed Whatever (#1234)
* fixed some more (#12345)
```

Which is uhhh, *a terrible hack* and makes me feel uncomfortable. I can try to submit a patch if you'd like, but I'm just curious if this is relevant to the project's goals.

### PR DESCRIPTION
**Pull request by hynek**
I believe this fixes #128.

While working on this I've noticed a big problem: the order of the categories is always alphabetical!? This needs to be fixed.

### CODE REVIEW

File: docs/keep-a-changelog.rst
Comment (adiroiban) [inline docs/keep-a-changelog.rst]: For now is ok.
I expect that Markdown is a common use case, so we might want to have a predefined markdown template, that is maintained by the main dev team.

File: docs/keep-a-changelog.rst
Comment (adiroiban) [inline docs/keep-a-changelog.rst]: Previously the config was something like this

```
    [[tool.towncrier.type]]
        directory = "security"
        name = "Security Fixes"
        showcontent = true

    [[tool.towncrier.type]]
        directory = "feature"
        name = "New Features"
        showcontent = true
```

I guess the main issue is due to pytoml not using an ordered dict.

**Review by adiroiban — APPROVED** [review]
Thanks. Looks good.

I think we should merge this.

I am not using markdown, but I expect that this is a very common use case.
This is why, in the future we might want to incorporate the template in towcrier.

Maybe have some "predefined" templates, that when connfigured will automatically also set/overwrite the `underlines`, `title_format` and `issue_format`

So that you can have just

```
   [tool.towncrier]
   directory = "changelog.d"
   filename = "CHANGELOG.md"
   start_string = "<!-- towncrier release notes start -->\n"
   template = "markdown"
```

When towncrier will see a config with `markdown` as the value, will not search for that file, but instead use some predefined values, with a template distributed with towncrier.

File: docs/keep-a-changelog.rst
Comment (hynek) [reply docs/keep-a-changelog.rst]: Ooof yeah you're right. It used to be an array but is a dict now and tomli gives it alphabetically:

```python
{'tool': {'towncrier': {'directory': 'changelog.d',
                        'filename': 'CHANGELOG.md',
                        'fragment': {'added': {'name': 'Added'},
                                     'changed': {'name': 'Changed'},
                                     'deprecated': {'name': 'Deprecated'},
                                     'fixed': {'name': 'Fixed'},
                                     'removed': {'name': 'Removed'},
                                     'security': {'name': 'Security'}},
                        'issue_format': '[#{issue}](https://github.com/twisted/my-project/issues/{issue})',
                        'start_string': '<!-- towncrier release notes start '
                                        '-->\n',
                        'template': 'changelog.d/changelog_template.jinja',
                        'title_format': '## '
                                        '[{version}](https://github.com/twisted/my-project/tree/{version}) '
                                        '- {project_date}',
                        'underlines': ['', '', '']}}}
```

Comment (adiroiban) [reply docs/keep-a-changelog.rst]: For reference.
This was implemented here https://github.com/twisted/towncrier/pull/370/files

I think the old format is still supported.

----------

This looks like a bug for the new format.

Comment (adiroiban) [reply docs/keep-a-changelog.rst]: I am not sure what toml lib we a re using, but it looks like ordered dict are not a thing :)

https://github.com/uiri/toml/pull/2

and also not wanted in the general TOML spec

https://github.com/toml-lang/toml/issues/162

Comment (hynek) [reply docs/keep-a-changelog.rst]: In modern Pythons, dicts are _always_ ordered. This is a deliberate action by tomli based on the spec that you linked. I've opened #437 to track this.

**Comment by hynek** [comment]
Please don't merge until #438 is merged and I have adapted this to the array style – thanks!

**Comment by hynek** [comment]
I just realized this one is not dependent on docs being updated so here it is. I've re-done it using the array-style definition.

**Review by adiroiban — APPROVED** [review]
Thanks. All good.

`directory` instead of `file_extension` is ugly ... but this life :)

**Comment by SmileyChris** [comment]
I was working on better markdown integration a while back, just resurrected that stash and made a PR for it: #439

Might be a much easier way to handle markdown formatting...

**Pull request merged**
